src/main/java/co/rsk/federate/signing/hsm/message/PowHSMSignerMessage.java

```diff
@@ -77,6 +113,48 @@ public Sha256Hash getSigHash() {
         return sigHash;
     }
 
+    public JsonNode getMessageToSign(int version) {
+        ObjectNode messageToSend = new ObjectMapper().createObjectNode();
+        messageToSend.put(TX.getFieldName(), getBtcTransactionSerialized());
+        messageToSend.put(INPUT.getFieldName(), inputIndex);
+        if (version == HSM_VERSION_5) {
+            if (hasWitness()) {
+                populateWithSegwitValues(messageToSend);
+            } else {
+                populateWithLegacyValues(messageToSend);
+            }
+        }
+        return messageToSend;
+    }
+
+    private void populateWithLegacyValues(ObjectNode messageToSend) {
+        messageToSend.put(SIGHASH_COMPUTATION_MODE.getFieldName(), SIGHASH_LEGACY_MODE);
+    }
+
+    private void populateWithSegwitValues(ObjectNode messageToSend) {
+        messageToSend.put(SIGHASH_COMPUTATION_MODE.getFieldName(), SIGHASH_SEGWIT_MODE);
+
+        long outpointValue = getOutpointValueForInputIndex();
+        messageToSend.put(OUTPOINT_VALUE.getFieldName(), outpointValue);
+
+        TransactionWitness witness = btcTransaction.getWitness(inputIndex);
+        String encodedWitnessScript = Hex.toHexString(witness.getScriptBytes());
+        messageToSend.put(WITNESS_SCRIPT.getFieldName(), encodedWitnessScript);
+    }
+
+    private long getOutpointValueForInputIndex() {
+        return Optional.of(outpointValues)
+            .filter(values -> inputIndex >= 0 && inputIndex < outpointValues.size())
```

Comment:
The `values` variable is not being used here. I guess the `outpointValues.size()` it's doing its job, since they are supposed to be the same, right? Since `values` is already there, what if we make use of it?

Reply:
Good catch

Reply:
Done